### ISSUE TITLE
xflux-gui is broken

### ISSUE THREAD

**Issue by dermetfan** [NixOS/nixpkgs#29055]
### Steps to reproduce

```
❯ nix-build '<nixpkgs>' -A xflux-gui
these derivations will be built:
  /nix/store/ph0ih2k0igbfwai2j1xrm4kxndmhdxkj-xflux-gui-2016-09-21.drv
building path(s) ‘/nix/store/yaq9qydskkfli67v3a3ba4c6g7sairss-xflux-gui-2016-09-21’
unpacking sources
unpacking source archive /nix/store/jdwpr0lx4i3l35jzwhmklmy0bbqmkplz-xflux-gui-0b56204477494b473a390e8b0db043437ec14f32-src
source root is xflux-gui-0b56204477494b473a390e8b0db043437ec14f32-src
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file xflux-gui-0b56204477494b473a390e8b0db043437ec14f32-src/src/fluxgui/xfluxcontroller.py
patching sources
configuring
building
WARNING: if you are running 'python setup.py' manually, and not as
part of Debian package creation, then you need to install the 'xflux'
binary separately. You can do this by running

    python ./download-xflux.py

before running 'setup.py'.
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/fluxgui
copying src/fluxgui/xfluxcontroller.py -> build/lib/fluxgui
copying src/fluxgui/settings.py -> build/lib/fluxgui
copying src/fluxgui/fluxcontroller.py -> build/lib/fluxgui
copying src/fluxgui/fluxapp.py -> build/lib/fluxgui
copying src/fluxgui/exceptions.py -> build/lib/fluxgui
copying src/fluxgui/__init__.py -> build/lib/fluxgui
copying src/fluxgui/preferences.glade -> build/lib/fluxgui
running build_scripts
creating build/scripts-2.7
copying and adjusting fluxgui -> build/scripts-2.7
changing mode of build/scripts-2.7/fluxgui from 644 to 755
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/fluxgui
copying build/lib/fluxgui/preferences.glade -> build/bdist.linux-x86_64/wheel/fluxgui
copying build/lib/fluxgui/__init__.py -> build/bdist.linux-x86_64/wheel/fluxgui
copying build/lib/fluxgui/exceptions.py -> build/bdist.linux-x86_64/wheel/fluxgui
copying build/lib/fluxgui/fluxapp.py -> build/bdist.linux-x86_64/wheel/fluxgui
copying build/lib/fluxgui/fluxcontroller.py -> build/bdist.linux-x86_64/wheel/fluxgui
copying build/lib/fluxgui/settings.py -> build/bdist.linux-x86_64/wheel/fluxgui
copying build/lib/fluxgui/xfluxcontroller.py -> build/bdist.linux-x86_64/wheel/fluxgui
running install_data
creating build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data
creating build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/data
creating build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/data/share
creating build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/data/share/icons
creating build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/data/share/icons/hicolor
creating build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/data/share/icons/hicolor/scalable
creating build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/data/share/icons/hicolor/scalable/apps
copying fluxgui.svg -> build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/data/share/icons/hicolor/scalable/apps
copying fluxgui-light.svg -> build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/data/share/icons/hicolor/scalable/apps
copying fluxgui-dark.svg -> build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/data/share/icons/hicolor/scalable/apps
creating build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/data/share/applications
copying desktop/fluxgui.desktop -> build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/data/share/applications
running install_egg_info
running egg_info
creating f.lux_indicator_applet.egg-info
writing f.lux_indicator_applet.egg-info/PKG-INFO
writing top-level names to f.lux_indicator_applet.egg-info/top_level.txt
writing dependency_links to f.lux_indicator_applet.egg-info/dependency_links.txt
writing manifest file 'f.lux_indicator_applet.egg-info/SOURCES.txt'
reading manifest file 'f.lux_indicator_applet.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching 'COPYING'
warning: no files found matching 'MANIFEST'
warning: no files found matching 'README'
warning: no files found matching 'xflux'
warning: no files found matching '*.desktop' under directory '*.svg'
warning: no files found matching '*.glade' under directory '*.py'
writing manifest file 'f.lux_indicator_applet.egg-info/SOURCES.txt'
Copying f.lux_indicator_applet.egg-info to build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8-py2.7.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/scripts
copying build/scripts-2.7/fluxgui -> build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/scripts
changing mode of build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.data/scripts/fluxgui to 755
creating build/bdist.linux-x86_64/wheel/f.lux_indicator_applet-1.1.8.dist-info/WHEEL
glibPreInstallPhase
installing
/tmp/nix-build-xflux-gui-2016-09-21.drv-0/xflux-gui-0b56204477494b473a390e8b0db043437ec14f32-src/dist /tmp/nix-build-xflux-gui-2016-09-21.drv-0/xflux-gui-0b56204477494b473a390e8b0db043437ec14f32-src
Processing ./f.lux_indicator_applet-1.1.8-py2-none-any.whl
Installing collected packages: f.lux-indicator-applet
Successfully installed f.lux-indicator-applet-1.1.8
/tmp/nix-build-xflux-gui-2016-09-21.drv-0/xflux-gui-0b56204477494b473a390e8b0db043437ec14f32-src
glibPreFixupPhase
post-installation fixup
Wrapping program /nix/store/yaq9qydskkfli67v3a3ba4c6g7sairss-xflux-gui-2016-09-21/bin/fluxgui
shrinking RPATHs of ELF executables and libraries in /nix/store/yaq9qydskkfli67v3a3ba4c6g7sairss-xflux-gui-2016-09-21
stripping (with flags -S) in /nix/store/yaq9qydskkfli67v3a3ba4c6g7sairss-xflux-gui-2016-09-21/lib  /nix/store/yaq9qydskkfli67v3a3ba4c6g7sairss-xflux-gui-2016-09-21/bin 
patching script interpreter paths in /nix/store/yaq9qydskkfli67v3a3ba4c6g7sairss-xflux-gui-2016-09-21
checking for references to /tmp/nix-build-xflux-gui-2016-09-21.drv-0 in /nix/store/yaq9qydskkfli67v3a3ba4c6g7sairss-xflux-gui-2016-09-21...
wrapping `/nix/store/yaq9qydskkfli67v3a3ba4c6g7sairss-xflux-gui-2016-09-21/bin/.fluxgui-wrapped'...
wrapping `/nix/store/yaq9qydskkfli67v3a3ba4c6g7sairss-xflux-gui-2016-09-21/bin/..fluxgui-wrapped-wrapped'...

Builder called die: makeWrapper doesn't understand the arg GI_TYPELIB_PATH
Backtrace:
99 makeWrapper /nix/store/1j8pmifn0v81w57imf8ai2rc5k0v1zsp-hook/nix-support/setup-hook
143 wrapProgram /nix/store/1j8pmifn0v81w57imf8ai2rc5k0v1zsp-hook/nix-support/setup-hook
100 wrapPythonProgramsIn /nix/store/zkzvyywmpxb8hrb26i474kqvpzmpavdx-hook/nix-support/setup-hook
4 wrapPythonPrograms /nix/store/zkzvyywmpxb8hrb26i474kqvpzmpavdx-hook/nix-support/setup-hook
102 _callImplicitHook /nix/store/r071pbl8rdb65cjlv55h67ds24pllank-stdenv/setup
113 _eval /nix/store/r071pbl8rdb65cjlv55h67ds24pllank-stdenv/setup
40 runHook /nix/store/r071pbl8rdb65cjlv55h67ds24pllank-stdenv/setup
940 fixupPhase /nix/store/r071pbl8rdb65cjlv55h67ds24pllank-stdenv/setup
1048 genericBuild /nix/store/r071pbl8rdb65cjlv55h67ds24pllank-stdenv/setup
2 main /nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh

builder for ‘/nix/store/ph0ih2k0igbfwai2j1xrm4kxndmhdxkj-xflux-gui-2016-09-21.drv’ failed with exit code 1
error: build of ‘/nix/store/ph0ih2k0igbfwai2j1xrm4kxndmhdxkj-xflux-gui-2016-09-21.drv’ failed
```

## Technical details

* System: `18.03pre114421.799435b7ca (Impala)`
* Nix version: `1.11.14`
* Nixpkgs version: `18.03pre114421.799435b7ca`
* Sandboxing enabled: no

**Comment by Ma27** [comment]
(triage) The build appears to be fixed now on `release-19.03` and `master`. However the startup fails with the following error:

```
Failed to contact configuration server; the most common cause is a missing or misconfigured D-Bus session bus daemon. See http://projects.gnome.org/gconf/ for information. (Details -  1: GetIOR failed: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.GConf was not provided by any .service files)
Critical error. Exiting.
Traceback (most recent call last):
  File "/nix/store/s2q75z6kbs589nnmi58ah4i1nv09wbi8-xflux-gui-1.1.10/bin/...fluxgui-wrapped-wrapped-wrapped", line 22, in <module>
    main()
  File "/nix/store/s2q75z6kbs589nnmi58ah4i1nv09wbi8-xflux-gui-1.1.10/lib/python2.7/site-packages/fluxgui/fluxapp.py", line 213, in main
    try:
  File "/nix/store/s2q75z6kbs589nnmi58ah4i1nv09wbi8-xflux-gui-1.1.10/lib/python2.7/site-packages/fluxgui/fluxapp.py", line 28, in __init__
    print "Critical error. Exiting."
  File "/nix/store/s2q75z6kbs589nnmi58ah4i1nv09wbi8-xflux-gui-1.1.10/lib/python2.7/site-packages/fluxgui/fluxapp.py", line 43, in exit
    try:
AttributeError: 'FluxGUI' object has no attribute 'xflux_controller'
Exception AttributeError: AttributeError("'FluxGUI' object has no attribute 'xflux_controller'",) in <bound method FluxGUI.__del__ of <fluxgui.fluxapp.FluxGUI object at 0x7f3b1bea3810>> ignored
```

I'm not exactly sure how to fix that though.

Tested locally at 3a4ffdd38b56801ce616aa08791121d36769e884 on my desktop (`none`+`i3wm`) and in a `gnome3` VM.

**Comment by JohnAZoidberg** [comment]
Fixed with eb5497c4199464edb6a5ea12dd5ff996af1cc021
